scripts/analytics/trialClassAverageCompletion.js

```diff
@@ -0,0 +1,234 @@
+// Find average classroom completion rate for first week of cs1 activity
+
+// Algorithm
+// 1. Find initial classroom student activity for first cs1 level between start and midpoint times
+// 2. Find course instances for newly active students
+// 3. Find free teachers for student course instances
+// 4. Find first non-empty cs1 course instances for teachers
+// 5. Find per-classroom cs1 course completion averages
+// 6. Compute overall average
+
+// NOTE: update the endDay variable below to control the date range
+// NOTE: assumes a couple unchanged course and first level Ids
+
+// TODO: use lodash
+// TODO: better handling of archived classrooms
+// TODO: corner cases around classroom membership changes not reflected in course instances
```

Comment:
Seems like classroom deleted members are not removed form course instances, but I did not investigate further.